markdown.dtx

```diff
@@ -25329,6 +25649,73 @@ end
 %    \end{macrocode}
 % \begin{markdown}
 %
+%#### Tex Math Dollars
+%
+% The \luamdef{extensions.tex_math_dollars} function implements the Pandoc
+% tex_math_dollars syntax extension.
+%
+% \end{markdown}
+%  \begin{macrocode}
+M.extensions.tex_math_dollars = function()
+  return {
+    name = "built-in tex_math_dollars syntax extension",
+    extend_writer = function(self)
+%    \end{macrocode}
+% \par
+% \begin{markdown}
+%
+% Define \luamdef{writer->tex_math_dollars} as a function that will transform
+% a tex_math_dollars span `s` of input text to the output format.
+%
+% \end{markdown}
+%  \begin{macrocode}
+      function self.display_math(s)
+        if not self.is_writing then return "" end
+        return {"\\markdownRendererDisplayMath{",s,"}"}
+      end
+
+      function self.inline_math(s)
+        if not self.is_writing then return "" end
+        return {"\\markdownRendererInlineMath{",s,"}"}
+      end
+    end, extend_reader = function(self)
+      local parsers = self.parsers
+      local writer = self.writer
+
+      local function between(p, starter, ender)
+        return (starter * Ct(p * (p - ender)^0) * ender)
+      end
+
+      parsers.inlinemathtail  = B(parsers.any * parsers.nonspacechar
+                                + parsers.backslash * parsers.any)
+                              * parsers.dollar 
+                              * -#(parsers.digit)
+
+      parsers.inlinemath = between(C(parsers.backslash^-1 * parsers.any
+                                   - parsers.blankline^2 - parsers.dollar),
+                                   parsers.dollar * #(parsers.nonspacechar),
+                                   parsers.inlinemathtail)
+
+      parsers.displaymathdelim = parsers.dollar * parsers.dollar
+
+      parsers.displaymath     = between(C(parsers.backslash^-1 * parsers.any 
+                                          - parsers.blankline^2 - parsers.dollar),
```

Comment:
I suggest the following changes that make the code more in line with the project's coding style.

```suggestion
      parsers.inlinemathtail  = B( parsers.any * parsers.nonspacechar
                                 + parsers.backslash * parsers.any)
                              * parsers.dollar 
                              * -#(parsers.digit)

      parsers.inlinemath = between(C( parsers.backslash^-1
                                    * parsers.any
                                    - parsers.blankline^2
                                    - parsers.dollar),
                                   parsers.dollar * #(parsers.nonspacechar),
                                   parsers.inlinemathtail)

      parsers.displaymathdelim = parsers.dollar
                               * parsers.dollar

      parsers.displaymath     = between(C( parsers.backslash^-1
                                         * parsers.any 
                                         - parsers.blankline^2
                                         - parsers.dollar),
```